packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -242,10 +242,22 @@ contract SystemConfig is OwnableUpgradeable, ISemver {
     }
 
     /// @notice Getter for the OptimismMintableERC20Factory address.
-    function optimismMintableERC20Factory() external view returns (address addr_) {
+    function optimismMintableERC20Factory() public view returns (address addr_) {
         addr_ = Storage.getAddress(OPTIMISM_MINTABLE_ERC20_FACTORY_SLOT);
     }
 
+    /// @notice Consolidated getter for the Addresses struct.
+    function getAddresses() external view returns (Addresses memory) {
+        return Addresses({
+            l1CrossDomainMessenger: l1CrossDomainMessenger(),
+            l1ERC721Bridge: l1ERC721Bridge(),
+            l1StandardBridge: l1StandardBridge(),
+            disputeGameFactory: disputeGameFactory(),
```

Comment:
Let's check about adding the ASR to this. We don't need to pass it as a new input because we can read it by going to DGF -> games -> ASR

Reply:
Discussions elsewhere made it clear that the ASR is a fluid thing, ie. there may be a single ASR for the superchain one day. We are able to discover it as is, so will not do this.